zerver/lib/string_validation.py

```diff
@@ -0,0 +1,37 @@
+import unicodedata
```

Comment:
Do we actually want a new file for these, rather than putting them in `zerver/lib/validators.py`?

Reply:
I believe that @timabbott was arguing for its own file earlier in this PR: https://github.com/zulip/zulip/pull/20375#discussion_r766040701